packages/EasyStandard/src/Rector/SingleLineCommentRector.php

```diff
@@ -0,0 +1,172 @@
+        }
+
+        return $node;
+    }
+
+    /**
+     * @param \PhpParser\Comment[] $comments
+     *
+     * @return \PhpParser\Comment[]
+     */
+    private function checkComments(array $comments): array
+    {
+        $newComments = [];
+        $isMultilineComment = false;
+
+        foreach ($comments as $index => $comment) {
+            $oldCommentText = $comment->getText();
+            if (Strings::startsWith($oldCommentText, '/*')) {
+                $newComments[] = $comment;
+                continue;
+            }
+
+            $commentText = Strings::trim(Strings::replace($oldCommentText, '#^\/\/#', ''));
+
+            if ($isMultilineComment === false && $this->isShouldBeStartFromLowerCase($commentText) === false) {
+                $commentText = Strings::firstUpper($commentText);
+            }
+
+            if (isset($comments[$index + 1])) {
+                $nextCommentText = $comments[$index + 1];
+                if ($nextCommentText !== '') {
+                    $isMultilineComment = true;
+                }
+            }
+
+            if (isset($comments[$index + 1]) === false) {
+                $isMultilineComment = false;
+            }
+
+            if ($isMultilineComment) {
+                $comment = $this->getNewCommentIfChanged($comment, '// ' . $commentText);
+                $newComments[] = $comment;
+                continue;
+            }
+
+            $disallowEnding = $this->checkLineEndingDisallowed($commentText);
+
+            if ($disallowEnding !== null) {
+                $pattern = '#' . \preg_quote($disallowEnding, null) . '$#';
+                $commentText = Strings::replace($commentText, $pattern, '');
+            }
+
+            $comment = $this->getNewCommentIfChanged($comment, '// ' . $commentText);
+
+            $newComments[] = $comment;
+        }
+
+        return $newComments;
+    }
+
+    private function checkLineEndingDisallowed(string $docLineContent): ?string
+    {
+        $result = null;
+
+        foreach ($this->disallowedEnd as $value) {
+            $isLineEndingWithDisallowed = Strings::endsWith($docLineContent, $value);
+            if ($isLineEndingWithDisallowed) {
+                $result = $value;
+                break;
+            }
+        }
+
+        return $result;
+    }
+
+    private function getNewCommentIfChanged(Comment $comment, string $commentText): Comment
+    {
+        if ($comment->getText() !== $commentText) {
+            $comment = new Comment(
+                $commentText,
+                $comment->getStartLine(),
+                $comment->getStartFilePos(),
+                $comment->getStartTokenPos(),
+                $comment->getEndLine(),
+                $comment->getEndFilePos(),
+                $comment->getEndTokenPos()
+            );
+        }
+
+        return $comment;
+    }
+
+    private function isShouldBeStartFromLowerCase(string $docLineContent): bool
```

Comment:
And this method should be named `isCommentIgnored`.

Reply:
Reworked.